### PR TITLE
[Dataloading] Ignore edge types without reverse types in edge dataloader

### PR DESCRIPTION
**Pull request by kkranen**
## Description
Fixes https://github.com/dmlc/dgl/issues/5410

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

**Comment by TristonC** [comment]
LGTM

**Comment by BarclayII** [comment]
@dgl-bot

**Review by BarclayII — APPROVED** [review]

**Comment by BarclayII** [comment]
@dgl-bot

**Comment by jermainewang** [comment]
Hi @kkranen , could you also help add a unit test?

**Comment by kkranen** [comment]
@jermainewang this test should be sufficient.

**Comment by jermainewang** [comment]
Thanks @kkranen ! Look like someone else has also changed the file. Would you please resolve the conflicts too? We will merge it afterwards.

**Comment by kkranen** [comment]
@jermainewang done.

**Comment by jermainewang** [comment]
@dgl-bot

File: tests/python/pytorch/dataloading/test_dataloader.py [hunk 1]
```diff
@@ -622,6 +622,36 @@ def test_edge_dataloader_excludes(
             break
 
 
+def test_edge_dataloader_exclusion_without_all_reverses():
```
Comment (jermainewang) [inline tests/python/pytorch/dataloading/test_dataloader.py:625]: Please fix the lint.

File: tests/python/pytorch/dataloading/test_dataloader.py [hunk 2]
```diff
@@ -622,6 +622,36 @@ def test_edge_dataloader_excludes(
...
+            )
+    d = dgl.dataloading.DataLoader(
+                graph=g,
+                indices={'AB':torch.tensor([0]),'BC':torch.tensor([0]),},
+                graph_sampler=block_sampler,
+                batch_size=2,
+                shuffle=True,
+                drop_last=False,
+                num_workers=0,
+                device='cuda',
```
Comment (jermainewang) [inline tests/python/pytorch/dataloading/test_dataloader.py:648]: The test is for CUDA only. My understanding is that this change applies to both CPU and GPU, so consider set the device according to the global context.

**Comment by jermainewang** [comment]
@Rhett-Ying Could you add @kkranen to the authorized users?

**Comment by Rhett-Ying** [comment]
@jermainewang  added.

**Comment by kkranen** [comment]
@dgl-bot

**Comment by kkranen** [comment]
@jermainewang @BarclayII just waiting on final approval now.

**Review by jermainewang — APPROVED** [review]
LGTM

**Pull request merged**